src/Microsoft.AspNetCore.Blazor.Browser/Interop/MethodInstance.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Blazor.Browser.Interop
+{
+    internal class MethodInstance
+    {
+        public MethodInstance()
+        {
+        }
+
+        public string Name { get; set; }
+        /// <summary>
+        /// Required if the method is generic.
+        /// </summary>
+        public IDictionary<string, TypeInstance> TypeArguments { get; set; }
+
+        /// <summary>
+        /// Required if the method has overloads.
+        /// </summary>
+        public TypeInstance[] ParameterTypes { get; set; }
+
+        internal MethodInfo GetMethodOrThrow(Type type)
+        {
+            var result = type.GetMethods(BindingFlags.Static | BindingFlags.Public).Where(m => string.Equals(m.Name, Name, StringComparison.Ordinal)).ToArray();
+
+            if (result.Length == 1)
+            {
+                // The method doesn't have overloads, we just return the method found by name.
+                return result[0];
+            }
+
+            result = result.Where(r => r.GetParameters().Length == ParameterTypes.Length).ToArray();
```

Comment:
@rynowak I'm not sure. That would require us to do lookups to get `Type` objects for each of the parameter types. In turn, would that require the end user to give full assembly-qualified type names including strong-name keys, etc?

If this ends up getting complex, then I think it would be preferable to limit this API to calling methods that don't have multiple overloads. Conceptually that's no different from requiring the developer to specify which overload they want, it just means they specify this information in the form of a wrapper method that calls what they want (and then get proper intellisense and compile-time checking that the thing they're calling actually exists).

Reply:
This tries to progressively increase the complexity.
If you don't have overloads there's no need to specify anything.
If there are overloads you have to indicate the types. 

As discussed offline, we wont document this for now.